http2/src/main/java/com/blazemeter/jmeter/http2/sampler/HTTP2Request.java

```diff
@@ -19,521 +27,529 @@
+  private static final String NON_HTTP_RESPONSE_MESSAGE = "Non HTTP response message";
+  public static final String IP_SOURCE = "HTTP2Request.ipSource"; // $NON-NLS-1$
+  public static final String IP_SOURCE_TYPE = "HTTP2Request.ipSourceType"; // $NON-NLS-1$
+  // Embedded URLs must match this regex (if provided)
+  public static final String EMBEDDED_URL_REGEX = "HTTPSampler.embedded_url_re"; // $NON-NLS-1$
+  // Store MD5 hash instead of storing response
+  public static final String MD5 = "HTTPSampler.md5"; // $NON-NLS-1$
+  public static final String EMBEDDED_RESOURCES = "HTTPSampler.embedded_resources"; // $NON-NLS-1$
+  public static final int SOURCE_TYPE_DEFAULT = HTTPSamplerBase.SourceType.HOSTNAME.ordinal();
+  public static final String ARGUMENTS = "HTTP2Request.Arguments"; // $NON-NLS-1$
+  public static final String POST_BODY_RAW = "HTTP2Request.postBodyRaw"; // TODO - belongs elsewhere
+  public static final boolean POST_BODY_RAW_DEFAULT = false;
+  public static final String DOMAIN = "HTTP2Request.domain"; // $NON-NLS-1$
+  public static final String RESPONSE_TIMEOUT = "HTTP2Request.response_timeout"; // $NON-NLS-1$
+  public static final String FOLLOW_REDIRECTS = "HTTP2Request.follow_redirects"; // $NON-NLS-1$
+  public static final String AUTO_REDIRECTS = "HTTP2Request.auto_redirects"; // $NON-NLS-1$
+  public static final String SYNC_REQUEST = "HTTP2Request.sync_request"; // $NON-NLS-1$
+  public static final String PROTOCOL = "HTTP2Request.protocol"; // $NON-NLS-1$
+  public static final String REQUEST_ID = "HTTP2Request.request_id"; // $NON-NLS-1$
+  /**
+   * This is the encoding used for the content, i.e. the charset name, not the header
+   * "Content-Encoding"
+   */
+  public static final String CONTENT_ENCODING = "HTTP2Request.contentEncoding"; // $NON-NLS-1$
+  public static final String PATH = "HTTP2Request.path"; // $NON-NLS-1$
+
+  private static ThreadLocal<Map<String, HTTP2Connection>> connections = ThreadLocal
+      .withInitial(HashMap::new);
+
+  private HTTP2Connection http2Connection;
+
+  public HTTP2Request() {
+    setName("HTTP2 Request");
+  }
+
+  @Override
+  public void setName(String name) {
+    if (name != null) {
+      setProperty(TestElement.NAME, name);
+    }
+  }
+
+  @Override
+  public String getName() {
+    return getPropertyAsString(TestElement.NAME);
+  }
+
+  @Override
+  public SampleResult sample(Entry entry) {
+    return sample();
+  }
+
+  /**
+   * Perform a sample, and return the results
+   *
+   * @return results of the sampling
+   */
+  public SampleResult sample() {
+
+    JMeterContext threadContext = getThreadContext();
+    JMeterVariables threadVars = threadContext.getVariables();
+    int nbActiveThreadsInThreadGroup = threadContext.getThreadGroup().getNumberOfThreads();
+    int nbTotalActiveThreads = JMeterContextService.getNumberOfThreads();
+
+    HTTP2SampleResult sampleResult = new HTTP2SampleResult(getName(), threadVars,
```

Comment:
One option to avoid passing too many parameters, can be to just pass the threadContext to the constructor and let the constructor extract the variables that it needs.

Reply:
that was the first approach but in the cases where the test finished and the response did not finish the context   does not have the values set